interpreters/src/validator.rs

```diff
@@ -0,0 +1,95 @@
+// Copyright 2022 CeresDB Project Authors. Licensed under Apache-2.0.
+
+use sql::plan::{Plan, ShowPlan};
+use table_engine::partition;
+
+use crate::interpreter::{PermissionDenied, Result};
+
+macro_rules! is_sub_table {
+    ($table_name:expr) => {{
+        let table_name = $table_name;
+        partition::is_sub_partition_table(table_name)
+    }};
+}
+
+/// Validator for [Plan]
+#[derive(Debug)]
+pub(crate) struct Validator {
+    ctx: ValidateContext,
+}
+
+impl Validator {
+    pub fn new(ctx: ValidateContext) -> Self {
+        Self { ctx }
+    }
+
+    pub fn validate(&self, plan: &Plan) -> Result<()> {
+        self.validate_admin_permission(plan)?;
+
+        Ok(())
+    }
+
+    fn validate_admin_permission(&self, plan: &Plan) -> Result<()> {
+        // Only admin can operate the sub tables(table partition) directly.
+        if Validator::contains_sub_tables(plan) && !self.ctx.admin {
```

Comment:
Maybe we should change the order of the boolean expression because in the most of the cases, admin is `true`:
```
if !self.ctx.admin && Validator::contains_sub_tables(plan)
```